src/models/CardDetails.ts

```diff
@@ -0,0 +1,8 @@
+import Card from './Card';
+
+interface CardDetails {
+  nonce?:string;
```

Comment:
Are these fields optional?

Reply:
No it's not.

Reply:
If they are not optional, should we make these fields required?

Reply:
We had already add ? sign for require fields.